pkg/controller/deployment/deployment_controller.go

```diff
@@ -431,13 +431,14 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	}
 
 	d := obj.(*extensions.Deployment)
+
 	everything := unversioned.LabelSelector{}
 	if reflect.DeepEqual(d.Spec.Selector, &everything) {
 		dc.eventRecorder.Eventf(d, api.EventTypeWarning, "SelectingAll", "This deployment is selecting all pods. A non-empty selector is required.")
 		return nil
 	}
 
-	if d.Spec.Paused {
+	if d.Spec.Paused || d.DeletionTimestamp != nil {
```

Comment:
A deleted deployment is different from a paused deployment; these two should be separate. I think the best fix would be to stop requeueing the deployment when it is deleted.